test/hotspot/jtreg/runtime/cds/appcds/TestCommon.java

```diff
@@ -745,4 +745,33 @@ public static String filterOutLogs(String stdout) {
         }
         return sb.toString();
     }
+
+    public static void stdoutMustMatch(OutputAnalyzer a, OutputAnalyzer b) {
+        linesMustMatch(a.getStdout().split("\n"),
+                             b.getStdout().split("\n"));
+    }
+
+    public static void linesMustMatch(String a[], String b[]) {
+        int max = Math.max(a.length, b.length);
+        for (int i = 0; i < max; i++) {
+            if (!a[i].equals(b[i])) {
+                System.out.println("a:" + i + " " + a[i]);
+                System.out.println("b:" + i + " " + b[i]);
+                throw new RuntimeException("Output mismatch on line :" + i
+                                           + ", a=" + a[i]
+                                           + ", b=" + b[i]);
+            }
+        }
+        if (a.length > b.length) {
+            throw new RuntimeException("Output mismatch on line :" + max
+                                       + ", a=" + a[max]
+                                       + ", b=<none>");
+
+        }
+        if (a.length < b.length) {
+            throw new RuntimeException("Output mismatch on line :" + max
+                                       + ", a=<none>"
+                                       + ", b=" + b[max]);
+        }
```

Comment:
Should these checks be done before the "for" loop at line 756? Otherwise, could one of the array be indexed out-out-bound?

Reply:
Thanks for finding the bug. I have fixed it in version [2e6af77](https://github.com/openjdk/jdk/pull/10639/commits/2e6af772766dff3432a11b436924185e1b270626). I also tested locally with this code:

```
String[] a = {"1", "2"}, b = {"1"};
TestCommon.linesMustMatch(a, b);
```

which correctly produces this error:

```
java.lang.RuntimeException: Output mismatch on line 1: a=2, b=<none>
```